EIPS/eip-894.md

```diff
@@ -0,0 +1,39 @@
+## Preamble
+
+    EIP: 894
+    Title: Immutability Enforcement Proposal (IMP)
+    Author: Marius Kjærstad
+    Type: Meta
+    Status: Draft
+    Created: 2018-02-21
+
+
+## Definition
+The definition of bailouts in this EIP is state changes to the Ethereum blockchain, through a hardfork, that treat account holders in different ways.
```

Comment:
This is an unusable definition.  The word "bailout" is defined as an act of giving financial assistance to a failing business or economy to save them from collapse.  This is not what you're proposing with your definition.

"Different ways" is far to vague.  By this definition, putting precompiles on low-order addresses is a "bailout" since it's a state change done through a hard fork that effects an account in a "different" way than other accounts.

Reply:
This is the definition for **this** EIP.

Reply:
I would avoid altogether the term "bailout"

As @AtLeastSignificant says, the term has a specific meaning, a redefinition for the purpose of this EIP adds ambiguity. Also "different ways" makes the concept too vague and open to a wide interpretation & application. This is the core of the definition, like this it's unusable/difficult to apply.

Besides, "bailout" is too charged of a concept, especially in the Ethereum/Crypto environment.

Reply:
I prefer to call a spade a spade.

Reply:
But you're doing the exact opposite of that.  You're calling state changes a "bailout".  The "through a hardfork" clause doesn't meany anything, and "that treat account holders in different ways" is also too vague to be usable.

I think you might rephrase to:

"The definition of bailouts in this EIP is *protocol changes* that recover lost, stolen, or otherwise inaccessible funds in ways outside the core functionality of the current protocol".

I still don't like the word "bailouts" just because you shouldn't use words with existing definitions that don't relate to what you're redefining the word to, but at least that definition is usable.

Reply:
I disagree.

Reply:
"bailout" isn't the worst conflation I've ever seen, but I agree we may be able to find a better word. On that note, let's all try and be nice and helpful. I'd suggest the phrase "ad hoc recovery", but we might be able to do even better.

Reply:
@sandakersmann 

Disagree with what?  I think my suggested definition actually hits what it is you're concerned about. Is this what you disagree with?

Or are you disagreeing with the fact that bailout is a defined word, and your use of it is a bastardization of that existing definition?

I think @bwheeler96 suggestion of "ad hoc recovery" is better, but it's sort of redundant IMO. When would you ever have a recovery that isn't "ad hoc"? 

If I were to come up with a term that accurately describes what I think @sandakersmann is getting to, I would use "protocol irregular state changes for fund recovery".

Reply:
Bush called torture for enhanced interrogation, but it was still torture.

Reply:
@AtLeastSignificant in my view an "ad hoc" recovery is a recovery that pertains to making unusual network changes for a specific misuse of Ethereum technology, whereas a recovery that isn't "ad hoc" might be something like fixing a bug in the EVM that caused loss of funds, in which case the recovery may be warranted and the fixing of a bug would not be ad hoc in nature. The latter being extremely unlikely and the former would ideally be impossible.